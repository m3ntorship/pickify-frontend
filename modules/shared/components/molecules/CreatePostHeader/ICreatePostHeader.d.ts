import type { ITabGroup } from '../TabGroup/ITabGroup';

declare namespace ICreatePostHeader {
  export interface IProps extends ITabGroup.IProps {
    profilePic?: string;
  }
}
export { ICreatePostHeader };
