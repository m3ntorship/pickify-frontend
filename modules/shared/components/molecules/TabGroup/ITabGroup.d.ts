import type { EPostType } from '../../../types/postFeed/EPostType';
import type { ITab } from '../../atoms/Tab/ITab';

declare namespace ITabGroup {
  export interface IProps extends IData, ITab.IHandlers {
    checkedValue: string;
    onlyLabel?: boolean;
  }

  export interface IData {
    tabsData: ITabGroupData[];
  }

  export interface ITabGroupData {
    id: string;
    svg: JSX.Element;
    value: 'Image Poll' | 'Mini Survey' | 'Text Poll';
    postType: EPostType;
  }
}
export { ITabGroup };
