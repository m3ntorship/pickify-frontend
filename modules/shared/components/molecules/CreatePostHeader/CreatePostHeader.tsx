import React from 'react';
import type { FC, ReactElement } from 'react';
import type { ICreatePostHeader } from './ICreatePostHeader';
import Avatar from '../../atoms/Avatar/Avatar';
import TabGroup from '../TabGroup/TabGroup';

const CreatePostHeader: FC<ICreatePostHeader.IProps> = ({
  tabsData,
  checkedValue,
  onTabChangeHandler,
  profilePic,
}): ReactElement => {
  const avatarVariant: 'filled' | 'notFilled' = profilePic
    ? 'filled'
    : 'notFilled';

  return (
    <div className="flex items-center mb-4">
      <div className="mr-4">
        <Avatar variant={avatarVariant} profilePic={profilePic} size="medium" />
      </div>
      <div>
        <TabGroup
          changeValHandler={onTabChangeHandler}
          checkedValue={checkedValue}
          tabsData={tabsData}
        />
      </div>
    </div>
  );
};

export default CreatePostHeader;
