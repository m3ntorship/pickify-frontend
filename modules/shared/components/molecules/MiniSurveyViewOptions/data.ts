export const optionsGroups = {
  groups: [
    {
      id: '03644270-7171-4147-b5a1-4233ff547f7a',
      name: 'Group Name',
      media: [],
      options: [
        {
          id: 'eeb0a32a-1a3c-4c7f-9e50-dd3d3f70b53d',
          body: 'ali tramsy',
          media: [{ url: 'https://placeimg.com/640/480/any' }],
          vote_count: 10,
          voted: false,
        },
        {
          id: 'eeb0a32a-1a3c-4c7f-9e50-dd3d3f70b5aa',
          body: ' ahmed essam',
          media: [{ url: 'https://placeimg.com/640/480/any' }],
          vote_count: 20,
          voted: false,
        },
        {
          id: 'eeb0a32a-1a3c-4c7f-9e50-dd3d3f70b53c',
          body: ' omar gamal',
          media: [{ url: 'https://placeimg.com/640/480/any' }],
          vote_count: 30,
          voted: false,
        },
      ],
    },
  ],
};
