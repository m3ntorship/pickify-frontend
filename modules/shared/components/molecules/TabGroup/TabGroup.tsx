import React from 'react';
import type { FC, ReactElement } from 'react';
import Tab from '../../atoms/Tab/Tab';
import type { ITabGroup } from './ITabGroup';

/**
 * @Props tabsData, checkedValue setCheckedValue are required
 * @ShouldHaveArrayOfDataToPassItToTheComponent
 * @example :  import { tabGroupData } from '@modules/shared/components/molecules/TabGroup/data'; // this function should return an array of objects
 * const [checkedValue, setCheckedValue] = useState('Image Poll');
 * <TabGroup tabsData={tabGroupData()} checkedValue={checkedValue} setCheckedValue={setCheckedValue} />
 * */

const TabGroup: FC<ITabGroup.IProps> = (props): ReactElement => {
  const { tabsData, checkedValue, changeValHandler, onlyLabel } = props;

  return (
    <div className="flex items-center">
      {tabsData.map((tab) => (
        <div className="mr-xs last:mr-0" key={tab.id}>
          <Tab
            id={tab.id}
            svg={tab.svg}
            value={tab.value}
            changeValHandler={changeValHandler}
            checkedValue={checkedValue}
            onlyLabel={onlyLabel}
            content={tab.value}
          />
        </div>
      ))}
    </div>
  );
};

export default TabGroup;
