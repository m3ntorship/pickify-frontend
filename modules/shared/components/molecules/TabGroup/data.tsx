import React from 'react';
import { EPostType } from '../../../types/postFeed/EPostType';
import ImagePoll from '../../icons/imagePoll.svg';
import TextPoll from '../../icons/textPoll.svg';
import MiniSurvey from '../../icons/miniSurvey.svg';
import styles from '../../atoms/Tab/Tab.module.css';
import type { ITabGroup } from './ITabGroup';

export const tabGroupData = (): ITabGroup.ITabGroupData[] => {
  return [
    {
      id: '1',
      svg: <ImagePoll className={styles.icon} data-testid="image-poll-icon" />,
      value: 'Image Poll',
      postType: EPostType.ImagePoll,
    },
    {
      id: '2',
      svg: <TextPoll className={styles.icon} />,
      value: 'Text Poll',
      postType: EPostType.TextPoll,
    },
    {
      id: '3',
      svg: <MiniSurvey className={styles.icon} />,
      value: 'Mini Survey',
      postType: EPostType.MiniSurvey,
    },
  ];
};
