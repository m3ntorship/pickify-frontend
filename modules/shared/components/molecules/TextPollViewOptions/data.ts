export const apiDummyData = {
  caption: 'nalyzing Delaware Frozen',
  options_groups: {
    groups: [
      {
        id: '03644270-7171-4147-b5a1-4233ff547f7a',
        name: 'Group Name',
        options: [
          {
            id: 'eeb0a32a-1a3c-4c7f-9e50-dd3d3f70b53d',
            body: 'ali tramsi',
          },
          {
            id: 'eeb0a32a-1a3c-4c7f-9e50-dd3d3f70b5aa',
            body: ' ahmed essam',
          },
          {
            id: 'eeb0a32a-1a3c-4c7f-9e50-dd3d3f70b53c',
            body: ' omar gamal',
          },
        ],
      },
    ],
  },
};
