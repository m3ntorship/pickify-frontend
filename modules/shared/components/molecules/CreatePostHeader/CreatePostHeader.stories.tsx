import type { ReactElement } from 'react';
import type { Story } from '@storybook/react';
import CreatePostHeader from './CreatePostHeader';
import type { ICreatePostHeader } from './ICreatePostHeader';
import { postCreationTabGroupData } from '../TabGroup/data';

export default {
  title: 'Molecules/CreatePostHeader',
  component: CreatePostHeader,
  argTypes: {
    checkedValue: {
      options: ['Image Poll', 'Text Poll', 'Mini survey'],
      control: { type: 'select' },
    },
  },
};

const Template: Story<ICreatePostHeader.IProps> = (args): ReactElement => (
  <CreatePostHeader {...args} />
);
export const createPostHeader = Template.bind({});
createPostHeader.args = {
  tabsData: postCreationTabGroupData(),
  profilePic: 'https://source.unsplash.com/random',
};
