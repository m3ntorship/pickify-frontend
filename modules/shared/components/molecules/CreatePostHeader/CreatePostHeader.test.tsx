import React from 'react';
import * as renderer from 'react-test-renderer';
import CreatePostHeader from './CreatePostHeader';
import { tabGroupData } from '../TabGroup/data';

describe('CreatePostHeader', () => {
  it('should render three tabs with filled avatar', () => {
    const changeValHandler = jest.fn();

    const tree = renderer
      .create(
        <CreatePostHeader
          tabsData={tabGroupData()}
          checkedValue="Image Poll"
          changeValHandler={changeValHandler}
          profilePic="https://images.unsplash.com/photo-1620577610365-86c411bad78d?ixlib=rb-1.2.1&q=80&fm=jpg&crop=entropy&cs=tinysrgb&w=1080&fit=max"
        />,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render three tabs with notFilled avatar', () => {
    const changeValHandler = jest.fn();

    const tree = renderer
      .create(
        <CreatePostHeader
          tabsData={tabGroupData()}
          checkedValue="Image Poll"
          changeValHandler={changeValHandler}
        />,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
