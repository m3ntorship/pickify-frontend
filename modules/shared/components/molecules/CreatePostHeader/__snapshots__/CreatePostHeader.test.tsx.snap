// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreatePostHeader should render three tabs with filled avatar 1`] = `
<div
  className="flex items-center mb-4"
>
  <div
    className="mr-4"
  >
    <div
      className="wrapper medium"
    >
      <div
        style={
          Object {
            "boxSizing": "border-box",
            "display": "inline-block",
            "margin": 0,
            "maxWidth": "100%",
            "overflow": "hidden",
            "position": "relative",
          }
        }
      >
        <div
          style={
            Object {
              "boxSizing": "border-box",
              "display": "block",
              "maxWidth": "100%",
            }
          }
        >
          <img
            alt=""
            aria-hidden={true}
            role="presentation"
            src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNDAiIGhlaWdodD0iNDAiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
            style={
              Object {
                "border": "none",
                "display": "block",
                "margin": 0,
                "maxWidth": "100%",
                "padding": 0,
              }
            }
          />
        </div>
        <img
          className="absolute w-full h-full"
          decoding="async"
          onClick={[Function]}
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style={
            Object {
              "border": "none",
              "bottom": 0,
              "boxSizing": "border-box",
              "display": "block",
              "height": 0,
              "left": 0,
              "margin": "auto",
              "maxHeight": "100%",
              "maxWidth": "100%",
              "minHeight": "100%",
              "minWidth": "100%",
              "objectFit": undefined,
              "objectPosition": undefined,
              "padding": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "visibility": "hidden",
              "width": 0,
            }
          }
        />
      </div>
    </div>
  </div>
  <div>
    <div
      className="flex items-center"
    >
      <div
        className="mr-xs last:mr-0"
      >
        <div
          className="flex"
        >
          <label
            className="tab-default tab-checked"
            data-testid="tab"
            data-value="image poll"
            htmlFor="1"
          >
            <div
              className="radio_container"
            >
              <input
                className="radio_input size_small"
                data-testid="Radio"
                defaultChecked={true}
                disabled={false}
                id="1"
                name="tab groub"
                onChange={[MockFunction]}
                type="radio"
                value="image poll"
              />
              <div
                className="radio_background unchecked size_small checkmark_small"
              />
            </div>
            <svg
              className="icon"
              data-file-name="SvgImagePoll"
              data-testid="image-poll-icon"
            />
            <span
              className="tab-type"
            >
              Image Poll
            </span>
          </label>
        </div>
      </div>
      <div
        className="mr-xs last:mr-0"
      >
        <div
          className="flex"
        >
          <label
            className="tab-default tab-unchecked"
            data-testid="tab"
            data-value="text poll"
            htmlFor="2"
          >
            <div
              className="radio_container"
            >
              <input
                className="radio_input size_small"
                data-testid="Radio"
                defaultChecked={false}
                disabled={false}
                id="2"
                name="tab groub"
                onChange={[MockFunction]}
                type="radio"
                value="text poll"
              />
              <div
                className="radio_background unchecked size_small checkmark_small"
              />
            </div>
            <svg
              className="icon"
              data-file-name="SvgTextPoll"
            />
            <span
              className="tab-type"
            >
              Text Poll
            </span>
          </label>
        </div>
      </div>
      <div
        className="mr-xs last:mr-0"
      >
        <div
          className="flex"
        >
          <label
            className="tab-default tab-unchecked"
            data-testid="tab"
            data-value="mini survey"
            htmlFor="3"
          >
            <div
              className="radio_container"
            >
              <input
                className="radio_input size_small"
                data-testid="Radio"
                defaultChecked={false}
                disabled={false}
                id="3"
                name="tab groub"
                onChange={[MockFunction]}
                type="radio"
                value="mini survey"
              />
              <div
                className="radio_background unchecked size_small checkmark_small"
              />
            </div>
            <svg
              className="icon"
              data-file-name="SvgMiniSurvey"
            />
            <span
              className="tab-type"
            >
              Mini survey
            </span>
          </label>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`CreatePostHeader should render three tabs with notFilled avatar 1`] = `
<div
  className="flex items-center mb-4"
>
  <div
    className="mr-4"
  >
    <div
      className="wrapper medium"
    >
      <svg
        className="absolute w-full h-full"
        data-file-name="SvgAnonymous"
        height={40}
        onClick={[Function]}
        width={40}
      />
      <svg
        className="absolute w-full h-full"
        data-file-name="SvgNotFilled"
        height={40}
        onClick={[Function]}
        width={40}
      />
    </div>
  </div>
  <div>
    <div
      className="flex items-center"
    >
      <div
        className="mr-xs last:mr-0"
      >
        <div
          className="flex"
        >
          <label
            className="tab-default tab-checked"
            data-testid="tab"
            data-value="image poll"
            htmlFor="1"
          >
            <div
              className="radio_container"
            >
              <input
                className="radio_input size_small"
                data-testid="Radio"
                defaultChecked={true}
                disabled={false}
                id="1"
                name="tab groub"
                onChange={[MockFunction]}
                type="radio"
                value="image poll"
              />
              <div
                className="radio_background unchecked size_small checkmark_small"
              />
            </div>
            <svg
              className="icon"
              data-file-name="SvgImagePoll"
              data-testid="image-poll-icon"
            />
            <span
              className="tab-type"
            >
              Image Poll
            </span>
          </label>
        </div>
      </div>
      <div
        className="mr-xs last:mr-0"
      >
        <div
          className="flex"
        >
          <label
            className="tab-default tab-unchecked"
            data-testid="tab"
            data-value="text poll"
            htmlFor="2"
          >
            <div
              className="radio_container"
            >
              <input
                className="radio_input size_small"
                data-testid="Radio"
                defaultChecked={false}
                disabled={false}
                id="2"
                name="tab groub"
                onChange={[MockFunction]}
                type="radio"
                value="text poll"
              />
              <div
                className="radio_background unchecked size_small checkmark_small"
              />
            </div>
            <svg
              className="icon"
              data-file-name="SvgTextPoll"
            />
            <span
              className="tab-type"
            >
              Text Poll
            </span>
          </label>
        </div>
      </div>
      <div
        className="mr-xs last:mr-0"
      >
        <div
          className="flex"
        >
          <label
            className="tab-default tab-unchecked"
            data-testid="tab"
            data-value="mini survey"
            htmlFor="3"
          >
            <div
              className="radio_container"
            >
              <input
                className="radio_input size_small"
                data-testid="Radio"
                defaultChecked={false}
                disabled={false}
                id="3"
                name="tab groub"
                onChange={[MockFunction]}
                type="radio"
                value="mini survey"
              />
              <div
                className="radio_background unchecked size_small checkmark_small"
              />
            </div>
            <svg
              className="icon"
              data-file-name="SvgMiniSurvey"
            />
            <span
              className="tab-type"
            >
              Mini survey
            </span>
          </label>
        </div>
      </div>
    </div>
  </div>
</div>
`;
